app/src/main/java/org/fossasia/openevent/app/data/LoginModel.java

```diff
@@ -43,18 +48,20 @@ public LoginModel(IUtilModel utilModel, EventService eventService, IDatabaseRepo
 
         return eventService
                 .login(new Login(username, password))
-                .doOnNext(loginResponse -> utilModel.saveToken(loginResponse.getAccessToken()))
                 .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .doOnComplete(() ->
-                    databaseRepository.getAllItems(User.class)
-                    .any(user -> !user.getEmail().equals(username))
-                    .filter(differentUser -> differentUser)
-                    .subscribeOn(Schedulers.io())
-                    .subscribe(differentUser ->
-                        utilModel.deleteDatabase()
-                            .subscribe()
-                    ));
+                .observeOn(AndroidSchedulers.mainThread());
+    }
+
+    @Override
+    public Observable<Boolean> isPreviousUser() {
+        return Observable.combineLatest(eventService.getUser(getAuthorization()),
+            databaseRepository.getAllItems(User.class).switchIfEmpty(Observable.just(new User(""))).firstElement().toObservable(), this::areSameUsers)
+            .subscribeOn(Schedulers.io())
```

Comment:
Why fetching the user from EventService here. It should be loaded at login itself. I mean to say that presenter should not trigger the logic to handle the previous user. Only LoginModel should do it. The presenter should only get the login success or error